provider.go

```diff
@@ -74,15 +166,28 @@ func getAPI(hostAddr string, maxRetries int) (client.API, error) {
 func Provider() terraform.ResourceProvider {
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
-			"tunnel_address": &schema.Schema{
+			"driver": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
+				Default: "tunnel",
+				Description: "Driver to use to connect to Fleet. Can be tunnel or api.",
```

Comment:
Ahh, yeah it would be good to mimic more completely how they build the client. Do you want that refactor in this PR or another one?